src/gg.regression.unity.bots/Samples~/ThirdPersonDemoURP/Bots/QuickstartBot/BotEntryPoint.cs

```diff
@@ -49,7 +50,7 @@ public override void ProcessTick(RG rgObject)
             if (thisBot == null) return; // We have not fully spawned yet, so wait
 
             // If we are attacking, just wait, and don't move
-            if (thisBot.GetField<bool>("IsAttacking")) return;
+            if (thisBot.GetFields<RGStateEntity_PlayerAttack>().FirstOrDefault()?.IsAttacking == true) return;
```

Comment:
This seems more complicated than before? I wonder if there are some utilities and other functions we can add

Reply:
Actually, if I recall, there was some work on allowing typed access to these - is that the case yet?

Reply:
This is one of the new accessors.

Since every behaviour that has states is now under a field in the state instead of flattened... there is an extra layer of access required.  You can get to that layer by `typeName` (the old way), or by the actual type in C# using this new way.  When you look at the agent builder PR you'll see that it still does things 'the old way', but with this extra layer still accounted for.